File: app/backend/src/services/queueService.js
/**
 * SQS message producer.
 * Publishes jobs to the main queue; the Lambda worker consumes them.
 *
 * AWS services used:
 *   - aws_sqs_queue.main          (job queue)
 *   - aws_sqs_queue.dead_letter   (DLQ for failed messages)
 *   - aws_kms_key.main            (queue encryption)
 *   - aws_lambda_event_source_mapping.sqs_trigger (consumer side)
 */

const { SendMessageCommand, GetQueueAttributesCommand } = require('@aws-sdk/client-sqs');
const { sqs } = require('../config/aws');

const QUEUE_URL = process.env.SQS_QUEUE_URL;

async function enqueue(jobType, payload) {
  const body = JSON.stringify({ jobType, payload, enqueuedAt: new Date().toISOString() });

  await sqs.send(new SendMessageCommand({
    QueueUrl:    QUEUE_URL,
    MessageBody: body,
    MessageAttributes: {
      JobType: { DataType: 'String', StringValue: jobType },
    },
  }));

  console.log(`Enqueued job type=${jobType}`);
}

async function getQueueDepth() {
  const result = await sqs.send(new GetQueueAttributesCommand({
    QueueUrl:       QUEUE_URL,
    AttributeNames: ['ApproximateNumberOfMessages'],
  }));
  return parseInt(result.Attributes.ApproximateNumberOfMessages, 10);
}

module.exports = { enqueue, getQueueDepth };

File: app/backend/src/services/queue_service.py
import json
import logging
from typing import Any, Dict, Literal

from src.config.aws import SQS_QUEUE_URL, sqs_client

logger = logging.getLogger(__name__)

JobType = Literal["PROCESS_CLAIM", "WELCOME_EMAIL", "POLICY_RENEWAL_REMINDER"]


def enqueue_job(job_type: JobType, payload: Dict[str, Any]) -> str:
    """
    Send a job message to the SQS queue.
    Returns the SQS MessageId.
    """
    message_body = json.dumps({
        "job_type": job_type,
        "payload": payload,
    })

    if not SQS_QUEUE_URL:
        logger.warning("SQS_QUEUE_URL not configured — skipping job: %s", job_type)
        return "LOCAL_SKIP"

    response = sqs_client.send_message(
        QueueUrl=SQS_QUEUE_URL,
        MessageBody=message_body,
        MessageAttributes={
            "JobType": {
                "StringValue": job_type,
                "DataType": "String",
            }
        },
    )
    message_id = response.get("MessageId", "UNKNOWN")
    logger.info("Enqueued job %s with MessageId %s", job_type, message_id)
    return message_id
